### DISCUSSION TITLE
physionet2012_dataset.keys()

### DISCUSSION THREAD

**Discussion by WarkerBoundless** [WenjieDu/PyPOTS#513] [Q&A] [answered]
set_random_seed()

# Load the PhysioNet-2012 dataset
physionet2012_dataset = gene_physionet2012(artificially_missing_rate=0.1)

# Take a look at the generated PhysioNet-2012 dataset, you'll find that everything has been prepared for you,
# data splitting, normalization, additional artificially-missing values for evaluation, etc.
print(physionet2012_dataset.keys())
我在执行这段代码时打印出来的结果是：dict_keys(['n_classes', 'n_steps', 'n_features', 'scaler', 'train_X', 'train_y', 'train_ICUType', 'val_X', 'val_y', 'val_ICUType', 'test_X', 'test_y', 'test_ICUType', 'val_X_ori', 'test_X_ori'])。跟示例比少了一个'test_X_indicating_mask'，这个怎么回事啊，该怎么解决。

**Comment by WenjieDu** [comment]
Hi there 👋,

Thank you so much for your attention to PyPOTS! You can follow me on GitHub to receive the latest news of PyPOTS. If you find PyPOTS helpful to your work, please star⭐️ this repository. Your star is your recognition, which can help more people notice PyPOTS and grow PyPOTS community. It matters and is definitely a kind of contribution to the community.

I have received your message and will respond ASAP. Thank you for your patience! 😃

Best,
Wenjie

**Comment by WenjieDu** [comment, marked as answer]
The Colab tutorials were out of date. I just updated them.
刚更新了colab上的过期教程，请查看
 https://colab.research.google.com/drive/1HEFjylEy05-r47jRy0H9jiS_WhD0UWmQ

**Reply by WarkerBoundless** [reply]
非常感谢您的回复，我根据您的修改代码成功运行了，但是我的运行结果和您的有点不一样，我在自己本地运行的训练损失比较大。请问这应该怎么解决。
2024-09-14 15:51:46 [INFO]: SAITS initialized with the given hyperparameters, the number of trainable parameters: 720,182
2024-09-14 15:52:19 [INFO]: Epoch 001 - training loss: 0.7185, validation loss: 6.2343
2024-09-14 15:52:19 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch1_loss6.234254938612382.pypots
2024-09-14 15:52:52 [INFO]: Epoch 002 - training loss: 0.5362, validation loss: 6.2159
2024-09-14 15:52:52 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch2_loss6.215857745707035.pypots
2024-09-14 15:53:26 [INFO]: Epoch 003 - training loss: 0.4918, validation loss: 6.1926
2024-09-14 15:53:26 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch3_loss6.1925995777050655.pypots
2024-09-14 15:53:48 [INFO]: Epoch 004 - training loss: 0.4607, validation loss: 6.1674
2024-09-14 15:53:48 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch4_loss6.167405763020118.pypots
2024-09-14 15:54:08 [INFO]: Epoch 005 - training loss: 0.4392, validation loss: 6.1562
2024-09-14 15:54:08 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch5_loss6.156184640278419.pypots
2024-09-14 15:54:38 [INFO]: Epoch 006 - training loss: 0.4227, validation loss: 6.1428
2024-09-14 15:54:38 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch6_loss6.142761386185884.pypots
2024-09-14 15:55:12 [INFO]: Epoch 007 - training loss: 0.4075, validation loss: 6.1406
2024-09-14 15:55:12 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch7_loss6.140596550703049.pypots
2024-09-14 15:55:45 [INFO]: Epoch 008 - training loss: 0.3970, validation loss: 6.1349
2024-09-14 15:55:45 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch8_loss6.1348588223258655.pypots
2024-09-14 15:56:19 [INFO]: Epoch 009 - training loss: 0.3867, validation loss: 6.1282
2024-09-14 15:56:19 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch9_loss6.128191532442967.pypots
2024-09-14 15:56:52 [INFO]: Epoch 010 - training loss: 0.3795, validation loss: 6.1272
2024-09-14 15:56:52 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS_epoch10_loss6.12722786962986.pypots
2024-09-14 15:56:52 [INFO]: Finished training. The best model is from epoch#10.
2024-09-14 15:56:52 [INFO]: Saved the model to tutorial_results/imputation/saits\20240914_T155132\SAITS.pypots
Testing mean absolute error: 0.2445

**Reply by WenjieDu** [reply]
应该是环境的问题，我之前在colab上也遇到过这种问题，你现在是用的cuda设备是么？切换到cpu的环境然后重新运行试试，应该是不同的依赖导致的，我没有查出是哪个依赖的问题，估计是numpy或者torch的版本问题

**Reply by WarkerBoundless** [reply]
2024-09-14 15:51:32 [INFO]: No given device, using default device: cpu
我应在是在cpu上运行的。非常感谢您的回复和指导。

**Reply by WenjieDu** [reply]
不客气。你可以在自己本地的设备上运行试试，如果仍然有这样的问题，请把环境中依赖版本导出`pip freeze > requirements.txt`发给我。你可以加我微信，拉你进PyPOTS微信社区群。你也可以在github上follow我，及时接收pypots团队的后续开源动态 ;-）
